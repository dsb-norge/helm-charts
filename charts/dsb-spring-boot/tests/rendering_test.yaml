suite: test deployment
tests:
  - it: Minimal manifest should match snapshot
    templates:
      - deployment.yaml
      - podDisruptionBudget.yaml
      - service.yaml
      - serviceAccount.yaml
    set:
      image: nginx
      tag: latest
    asserts:
      - matchSnapshot: { }
  - it: should render default with PodDisruptionBudget and should match snapshot
    templates:
      - podDisruptionBudget.yaml
    asserts:
      - matchSnapshot: { }
      - isKind:
          of: PodDisruptionBudget
  - it: should be possible to render without PodDisruptionBudget
    templates:
      - podDisruptionBudget.yaml
    set:
      createPodDisruptionBudget: false
    asserts:
      - hasDocuments:
          count: 0
  - it: should be possible to render without ServiceMonitor
    templates:
      - serviceMonitor.yaml
    set:
      prometheus_enabled: "false"
    asserts:
      - hasDocuments:
          count: 0
  - it: Full manifest should match snapshot
    templates:
      - configMap.yaml
      - databaseDeployment.yaml
      - databaseService.yaml
      - deployment.yaml
      - ingress.yaml
      - podDisruptionBudget.yaml
      - service.yaml
      - serviceAccount.yaml
      - certificate-secret.yaml
      - serviceMonitor.yaml
      - clusterRole.yaml
      - clusterRoleBinding.yaml
      - mountedConfigMap.yaml
      - csi-secrets-store-azure.yaml
    set:
      replicas: 3
      minAvailableReplicas: 1
      image: wordpress
      tag: greatest
      imagePullPolicy: customValue
      deploymentAnnotations:
        key1: value1
        key2: value2
      springProfiles: "chart-testing"
      config:
        root:
          key1: value1
          key2: value2
      secretRefs:
        - secret1
      configMapRefs:
        - configMap1
      ingress_host: example.com
      ingress_path: /example
      application_web_port: 80
      application_actuator_port: 81
      prometheus_enabled: "true"
      prometheus_path: "/path/to/prometheus"
      memory_request: "4Mi"
      memory_limit: "19Mi"
      cpu_request: "0.3"
      cpu_limit: "1"
      java_opts: "-XX:MaxDirectMemorySize=256M -XX:MaxMetaspaceSize=128M"
      jvm_head_room: 12.34
      database_container:
        enabled: true
        image: "test-db:v1"
        database: "test-data"
        user: "custom user"
        password: "password123"
      dashboard:
        name: "Rendering test"
        component: "Backend API"
        partOf: "Verification stuff"
        createdBy: "https://github.com/dsb-norge/helm-charts"
      websocket:
        enabled: true
      rbac:
        enabled: true
        roles:
          podPermission:
            apiGroups:
              - ""
              - apps
            resources:
              - pods
              - deployments
            verbs:
              - list
              - watch
              - get
          nodePermissions:
            apiGroups:
              - ""
            resources:
              - nodes
            verbs:
              - list
              - get
      certificates:
        dsb-certificate:
          key_vault:
            name: azure_key_vault_name
            keystore_ref: vaule_for_certificate
            keystore_password_ref: value_for_password
            keystore_alias_ref: value_for_alias
          fileName: dsb-test-virksomhetssertifikat.pfx
      mountedConfig:
        ansible-ca:
          fileName: ansible-ca.pem
          content: |
            -----BEGIN CERTIFICATE-----
            MIIFFTCCAv2gAwIBAgIUB7/W9hZbOPR/Co/SFx9rlV71A+swDQYJKoZIhvcNAQEL
            -----END CERTIFICATE-----
      azureKeyVault:
        defaultValues:
          tenantId: b2255c28-9e28-4641-a6bc-b63391d59143
          clientId: b2255c28-9e28-4641-a6bc-b63391d59144
          kvName: kv-rg00-ss0-render-dev2
        vaults:
          myKv1:
            tenantId: b2255c28-9e28-4641-a6bc-b63391d59141
            clientId: b2255c28-9e28-4641-a6bc-b63391d59142
            kvName: kv-rg00-ss0-render-dev1
            keys:
              - nameInKv: "name-of-key-in-vault"
                mountAsEnv: "name.of.env.variable.holding.key.in.pem.format"
                fileMountPathEnvName: "name.of.env.variable.holding.path.to.key.as.file"
                fileMountPath: "/custom/file/paths/are/supported"
            secrets:
              - nameInKv: "name-of-secret-in-vault"
                fileMountPath: "/hardcoded/path/of/secret/as/file"
            certs:
              - nameInKv: "name-of-cert-in-vault"
                fileMountPathEnvName: "name.of.env.variable.holding.path.to.cert.as.file"
          myKv2:
            keys:
              - nameInKv: "name-of-key-in-vault"
                mountAsEnv: "key_in_2_var"
            secrets:
              - nameInKv: "name-of-secret-in-vault"
                fileMountPath: "/secret_in_2_file"
            certs:
              - nameInKv: "name-of-cert-in-vault"
                fileMountPathEnvName: "cert_in_2_file_path_var"
    asserts:
      - matchSnapshot: { }
