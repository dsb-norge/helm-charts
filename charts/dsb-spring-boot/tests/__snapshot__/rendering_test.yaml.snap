Full manifest should match snapshot:
  1: |
    apiVersion: v1
    data:
      root.key1: value1
      root.key2: value2
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME-configmap
      namespace: NAMESPACE
  2: |
    apiVersion: batch/v1
    kind: CronJob
    metadata:
      annotations:
        cronKey1: cronValue1
        cronKey2: cronValue2
      name: RELEASE-NAME-cron-cron-example
      namespace: NAMESPACE
    spec:
      concurrencyPolicy: Forbid
      failedJobsHistoryLimit: 3
      jobTemplate:
        metadata:
          annotations:
            co.elastic.logs/json.add_error_key: "true"
            co.elastic.logs/json.keys_under_root: "true"
            co.elastic.logs/json.overwrite_keys: "true"
        spec:
          activeDeadlineSeconds: 60
          backoffLimit: 3
          completions: 1
          parallelism: 1
          template:
            metadata:
              labels:
                app: RELEASE-NAME-cron
                cron: cron-example
            spec:
              containers:
              - env:
                - name: TOKEN_URL
                  value: https://example.com/token
                - name: CLIENT_ID
                  value: release-name-cron-client
                - name: API_METHOD
                  value: TRACE
                - name: API_PATH
                  value: http://RELEASE-NAME.NAMESPACE.svc.cluster.local:80/do
                envFrom:
                - secretRef:
                    name: jobSecret1
                image: dsbacr.azurecr.io/dsb-norge/oauth2-api-caller:greatest
                name: cron-example
                resources:
                  limits:
                    cpu: 50m
                    memory: 128Mi
                  requests:
                    cpu: 50m
                    memory: 64Mi
                securityContext:
                  allowPrivilegeEscalation: false
                  capabilities:
                    drop:
                    - ALL
                  privileged: false
                  readOnlyRootFilesystem: true
                  runAsGroup: 3000
                  runAsNonRoot: true
                  runAsUser: 1000
              restartPolicy: OnFailure
      schedule: '*/2 * * * *'
      successfulJobsHistoryLimit: 2
  3: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-database-deployment
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app: RELEASE-NAME-db-app
      template:
        metadata:
          annotations:
            checksum: chart-version=3.0.14_config-hash=a306457112a64a29a496501b3c650b192d3364a4671a301960ca83f838db924f
          labels:
            app: RELEASE-NAME-db-app
        spec:
          containers:
          - env:
            - name: ACCEPT_EULA
              value: "Y"
            - name: SA_PASSWORD
              value: password123
            - name: MSSQL_SA_PASSWORD
              value: password123
            image: test-db:v1
            name: RELEASE-NAME-database-container
            ports:
            - containerPort: 1433
            resources:
              limits:
                cpu: "2.5"
                memory: 2048Mi
              requests:
                cpu: "0.2"
                memory: 1024Mi
          hostname: RELEASE-NAME-db-svc
  4: |
    apiVersion: v1
    kind: Service
    metadata:
      name: RELEASE-NAME-db-svc
      namespace: NAMESPACE
    spec:
      ports:
      - port: 1433
        protocol: TCP
        targetPort: 1433
      selector:
        app: RELEASE-NAME-db-app
  5: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        key1: value1
        key2: value2
      name: RELEASE-NAME-deployment
      namespace: NAMESPACE
    spec:
      replicas: 3
      selector:
        matchLabels:
          app: RELEASE-NAME-app
      template:
        metadata:
          annotations:
            apparmor.security.beta.kubernetes.io/pod: runtime/default
            checksum: chart-version=3.0.14_config-hash=6413cae9d88b8807f26099edc976e1d2b59e9c1a0ad2b805fc56a67af63da0c0
            co.elastic.logs/json.add_error_key: "true"
            co.elastic.logs/json.keys_under_root: "true"
            co.elastic.logs/json.overwrite_keys: "true"
          labels:
            app: RELEASE-NAME-app
        spec:
          affinity:
            podAntiAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
              - labelSelector:
                  matchExpressions:
                  - key: app
                    operator: In
                    values:
                    - RELEASE-NAME-app
                topologyKey: kubernetes.io/hostname
          containers:
          - env:
            - name: SPRING_PROFILES_ACTIVE
              value: chart-testing
            - name: MANAGEMENT_METRICS_EXPORT_PROMETHEUS_ENABLED
              value: true
            - name: DATABASE_CONTAINER_USER
              value: sa
            - name: DATABASE_CONTAINER_PASSWORD
              value: password123
            - name: DATABASE_CONTAINER_HOST_AND_PORT
              value: RELEASE-NAME-db-svc:1433
            - name: DATABASE_CONTAINER_DATABASE
              value: emptydb
            - name: DSB_CERTIFICATE_CERTIFICATE_PATH
              value: /certificates/dsb-test-virksomhetssertifikat.pfx
            envFrom:
            - secretRef:
                name: secret1
            - secretRef:
                name: dsb-certificate-info
            - configMapRef:
                name: configMap1
            - configMapRef:
                name: RELEASE-NAME-configmap
            image: wordpress:greatest
            imagePullPolicy: customValue
            lifecycle:
              preStop:
                exec:
                  command:
                  - sh
                  - -c
                  - sleep 10
            livenessProbe:
              failureThreshold: 20
              httpGet:
                path: /actuator/health/liveness
                port: 81
              initialDelaySeconds: 15
              periodSeconds: 15
            name: RELEASE-NAME-container
            ports:
            - containerPort: 80
              name: web
            - containerPort: 81
              name: actuator
            readinessProbe:
              failureThreshold: 20
              httpGet:
                path: /actuator/health/readiness
                port: 81
              initialDelaySeconds: 10
              periodSeconds: 15
            resources:
              limits:
                cpu: 1
                memory: 19Mi
              requests:
                cpu: 0.3
                memory: 4Mi
            securityContext:
              allowPrivilegeEscalation: false
              capabilities:
                drop:
                - ALL
              privileged: false
              readOnlyRootFilesystem: true
              runAsGroup: 3000
              runAsNonRoot: true
              runAsUser: 1000
            volumeMounts:
            - mountPath: /tmp
              name: tmp-dir
            - mountPath: /certificates
              name: dsb-certificate
              readOnly: true
          nodeSelector:
            NodePool: workers
          serviceAccountName: RELEASE-NAME-service-account
          volumes:
          - emptyDir: {}
            name: tmp-dir
          - name: dsb-certificate
            secret:
              secretName: dsb-certificate-file
  6: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      rules:
      - host: example.com
        http:
          paths:
          - backend:
              service:
                name: RELEASE-NAME
                port:
                  number: 80
            path: /example
            pathType: Prefix
  7: |
    apiVersion: policy/v1
    kind: PodDisruptionBudget
    metadata:
      name: RELEASE-NAME.pdb
    spec:
      minAvailable: 1
      selector:
        matchLabels:
          app: RELEASE-NAME-app
  8: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app: RELEASE-NAME-service
        chart-name: dsb-spring-boot
        chart-version: 3.0.14
        management.port: "81"
        spring-boot: "true"
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      ports:
      - name: web
        port: 80
      - name: actuator
        port: 81
      selector:
        app: RELEASE-NAME-app
  9: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app: RELEASE-NAME-app
      name: RELEASE-NAME-service-account
      namespace: NAMESPACE
  10: |
    apiVersion: kubernetes-client.io/v1
    kind: ExternalSecret
    metadata:
      name: dsb-certificate-info
      namespace: NAMESPACE
    spec:
      backendType: azureKeyVault
      data:
      - key: value_for_password
        name: DSB_CERTIFICATE_CERTIFICATE_KEYSTORE
      - key: value_for_alias
        name: DSB_CERTIFICATE_CERTIFICATE_ALIAS
      keyVaultName: azure_key_vault_name
  11: |
    apiVersion: kubernetes-client.io/v1
    kind: ExternalSecret
    metadata:
      name: dsb-certificate-file
      namespace: NAMESPACE
    spec:
      backendType: azureKeyVault
      data:
      - isBinary: true
        key: vaule_for_certificate
        name: dsb-test-virksomhetssertifikat.pfx
      keyVaultName: azure_key_vault_name
  12: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      labels:
        release: kube-prometheus
      name: RELEASE-NAME-monitor
      namespace: NAMESPACE
    spec:
      endpoints:
      - path: /path/to/prometheus
        port: actuator
        scheme: http
      namespaceSelector:
        matchNames:
        - NAMESPACE
      selector:
        matchLabels:
          app: RELEASE-NAME-service
Minimal manifest should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-deployment
      namespace: NAMESPACE
    spec:
      replicas: 2
      selector:
        matchLabels:
          app: RELEASE-NAME-app
      template:
        metadata:
          annotations:
            apparmor.security.beta.kubernetes.io/pod: runtime/default
            checksum: chart-version=3.0.14_config-hash=ab6db3f6b66b182d4576eb437dfc5556ae4857369aef33e0d18dbe34756df906
            co.elastic.logs/json.add_error_key: "true"
            co.elastic.logs/json.keys_under_root: "true"
            co.elastic.logs/json.overwrite_keys: "true"
          labels:
            app: RELEASE-NAME-app
        spec:
          affinity:
            podAntiAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
              - labelSelector:
                  matchExpressions:
                  - key: app
                    operator: In
                    values:
                    - RELEASE-NAME-app
                topologyKey: kubernetes.io/hostname
          containers:
          - env:
            - name: SPRING_PROFILES_ACTIVE
              value: kubernetes
            envFrom: null
            image: nginx:latest
            lifecycle:
              preStop:
                exec:
                  command:
                  - sh
                  - -c
                  - sleep 10
            livenessProbe:
              failureThreshold: 20
              httpGet:
                path: /actuator/health/liveness
                port: 8180
              initialDelaySeconds: 15
              periodSeconds: 15
            name: RELEASE-NAME-container
            ports:
            - containerPort: 8080
              name: web
            - containerPort: 8180
              name: actuator
            readinessProbe:
              failureThreshold: 20
              httpGet:
                path: /actuator/health/readiness
                port: 8180
              initialDelaySeconds: 10
              periodSeconds: 15
            resources:
              limits:
                cpu: 2.5
                memory: 1024Mi
              requests:
                cpu: 100m
                memory: 256Mi
            securityContext:
              allowPrivilegeEscalation: false
              capabilities:
                drop:
                - ALL
              privileged: false
              readOnlyRootFilesystem: true
              runAsGroup: 3000
              runAsNonRoot: true
              runAsUser: 1000
            volumeMounts:
            - mountPath: /tmp
              name: tmp-dir
          nodeSelector:
            NodePool: workers
          serviceAccountName: RELEASE-NAME-service-account
          volumes:
          - emptyDir: {}
            name: tmp-dir
  2: |
    apiVersion: policy/v1
    kind: PodDisruptionBudget
    metadata:
      name: RELEASE-NAME.pdb
    spec:
      minAvailable: 1
      selector:
        matchLabels:
          app: RELEASE-NAME-app
  3: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app: RELEASE-NAME-service
        chart-name: dsb-spring-boot
        chart-version: 3.0.14
        management.port: "8180"
        spring-boot: "true"
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      ports:
      - name: web
        port: 8080
      - name: actuator
        port: 8180
      selector:
        app: RELEASE-NAME-app
  4: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app: RELEASE-NAME-app
      name: RELEASE-NAME-service-account
      namespace: NAMESPACE
should render default with PodDisruptionBudget and should match snapshot:
  1: |
    apiVersion: policy/v1
    kind: PodDisruptionBudget
    metadata:
      name: RELEASE-NAME.pdb
    spec:
      minAvailable: 1
      selector:
        matchLabels:
          app: RELEASE-NAME-app
